components/list/list-item-drag-mixin.js

```diff
@@ -24,96 +84,285 @@ export const ListItemDragMixin = superclass => class extends superclass {
+		e.preventDefault();
+	}
+
+	_onDragEnd(e) {
+		const dropSpot = getDragState();
+		this.dragging = false;
+		if (dropSpot.shouldDrop(e.timeStamp)) {
+			this._annoucePositionChange(dropSpot.dragTargetKey, dropSpot.dropTargetKey, dropSpot.dropLocation);
 		}
+		clearDragState();
 	}
 
-	moveAfter(itemKey) {
-		if (itemKey === undefined) {
-			return;
+	_onDragHandleActions(e) {
+		switch (e.detail.action) {
+			case dragActions.active:
+				this._keyboardActive = true;
+				break;
+			case dragActions.cancel:
+			case dragActions.save:
+				this._keyboardActive = false;
+				break;
+			case dragActions.up:
+				this._annoucePositionChange(this.key, null, dropLocation.shiftUp);
+				break;
+			case dragActions.down:
+				this._annoucePositionChange(this.key, null, dropLocation.shiftDown);
+				break;
+			case dragActions.first:
+				this._annoucePositionChange(this.key, null, dropLocation.first);
+				break;
+			case dragActions.last:
+				this._annoucePositionChange(this.key, null, dropLocation.last);
+				break;
+			default:
+				break;
 		}
-		this._dispatchDragEvent(itemKey);
+
 	}
 
-	moveBefore(itemKey) {
-		if (itemKey === undefined) {
-			return;
+	_onDragOver(e) {
+		if (!this.key) return;
+		const dropSpots = getDragState();
+		dropSpots.updateTime(e.timeStamp);
+		e.preventDefault();
+	}
+
+	_onDragStart(e) {
+		e.dataTransfer.setData('text/plain', `${this.dropText}`);
+		e.dataTransfer.effectAllowed = 'move';
+
+		//legacy edge doesn't support setDragImage. Experience is not degraded for legacy edge by doing this fix.
+		if (e.dataTransfer.setDragImage) {
+			const nodeImage = this.shadowRoot.querySelector('.d2l-list-item-drag-image') || this;
+			e.dataTransfer.setDragImage(nodeImage, 50, 50);
 		}
 
-		this._dispatchDragEvent(itemKey);
+		createDragState(this);
+
+		setTimeout(() => {
+			this.dragging = true;
+		});
 	}
 
-	// TODO
-	_copy() {
+	_onDrop() {
+		const dropSpots = getDragState();
+		dropSpots.setActiveDropTarget(this, dropSpots.dropLocation);
+	}
 
+	_onDropAreaBottomDrag(e) {
+		e.dataTransfer.dropEffect = 'move';
+		const dropSpots = getDragState();
+		dropSpots.setActiveDropTarget(this, dropLocation.below);
+		this._inBottomArea = true;
 	}
 
-	_dispatchDragEvent(detail) {
-		this.dispatchEvent(new CustomEvent('d2l-list-item-position', {
-			detail: detail,
-			bubbles: true
-		}));
+	_onDropAreaDragEnter(e) {
+		e.dataTransfer.dropEffect = 'move';
+		const dropSpots = getDragState();
+		dropSpots.setActiveDropTarget(this, dropLocation.above);
+		this._inTopArea = true;
 	}
 
-	_dragEnter() {
-		this._dragStartHandler();
+	_onDropAreaLowerDragEnter(e) {
+		e.dataTransfer.dropEffect = 'move';
+		if (this._inBottomArea) {
+			const dropSpots = getDragState();
+			dropSpots.setActiveDropTarget(this, dropLocation.above);
+			this._inBottomArea = false;
+		}
 	}
 
-	_dragExit() {
-		this._dispatchDragEvent(null);
+	_onDropAreaUpperDragEnter(e) {
```

Comment:
Sounds like we decided on "drop target" term... do these handlers and reference to "dropArea" in `render` also need to be updated?